.changelog/110.txt

```diff
@@ -0,0 +1,7 @@
+```release-note:breaking-change
+All `[]*tfprotov6.Diagnostic` handling has been converted to `diag.Diagnostics` except where interfacing directly with the underlying protocol type.
```

Comment:
Can we be more precise about what the breaking change is here? Specifically, is it that functions that used to return `[]*tfprotov6.Diagnostic` now return `diag.Diagnostics`?

Reply:
Happy to update as necessary! Some starter ideas:

1. All `[]*tfprotov6.Diagnostic` handling has been converted to `diag.Diagnostics` except where interfacing directly with the underlying protocol type. This includes all function signatures, which will require provider updates. Replacing `[]*tfprotov6.Diagnostic` with `diag.Diagnostics` and import path `github.com/hashicorp/terraform-plugin-go/tfprotov6` with `github.com/hashicorp/terraform-plugin-framework/diag` should suffice in most cases.
1. All `[]*tfprotov6.Diagnostic` handling has been converted to `diag.Diagnostics` except where interfacing directly with the underlying protocol type. Providers will require updates to function signatures and import paths.
1. All `[]*tfprotov6.Diagnostic` handling has been converted to `diag.Diagnostics` except where interfacing directly with the underlying protocol type. Providers will require updates to function signatures (`[]*tfprotov6.Diagnostic` to `diag.Diagnostics`) and import paths (`github.com/hashicorp/terraform-plugin-go/tfprotov6` to `github.com/hashicorp/terraform-plugin-framework/diag`).

Reply:
```suggestion
Most uses of `[]*tfprotov6.Diagnostic` have been replaced with a new `diag.Diagnostics` type. Please update your type signatures, and use one of the `diags.New*` helper functions instead of constructing `*tfprotov6.Diagnostic`s by hand.
```